wgsl/index.bs

```diff
@@ -375,23 +375,44 @@ Issue: (dneto) It feels like this needs some reorganization. Perhaps "Evaluation
 
 ## Scoping ## {#scoping}
 
-Scoping is the set of rules determining where a variable may be used.
+A declaration introduces a name, given by an identifier token.
+Scoping is the set of rules determining where that name may be used, in relation to
+the position of the declaration in the program.
+If a name may be used at a particular point in the program, then we say it is *in scope*.
 
 Issue: (dneto) also lifetime.
 
 There are multiple levels of scoping depending on how and where things are
 declared.
 
-Note: Shadow variables are not allowed in [SHORTNAME]. A variable must not be defined
-         in a given scope or any scope above the current one.
+A declaration must not introduce a name when that name is already in scope at the start
+of the declaration.
+That is, shadow names are not allowed in [SHORTNAME].
 
-## Module Scope ## {#module-scope}
-Any variable declared outside a function is at module scope.
+### Module Scope ### {#module-scope}
 
-## Function Scope ## {#function-scope}
-Variables declared within a function are at function scope. The variable is
-visible at any scoping level which comes after the variable declaration in the
```

Comment:
The "any scoping level which comes after the variable declaration in the source code" is not correct. It only extends until the end of the immediately-enclosing statement block.
We don't have a definition for "statement block" so I wrote it out in more words.